utilities/misc/randSample.m

```diff
@@ -0,0 +1,57 @@
+function outputSample = randSample(population, numItems, weights)
+  % Take a random sample of values.
+  % GPL-licensed replacement for `randsample` from Statistics and Machine
+  % Learning Toolbox.
+  % 
+  % Note that unweighted sampling is *without* replacement, and weighted
+  % sampling is *with* replacement!
+  % 
+  % Parameters
+  % ----------
+  % population : vector or int
+  %   if a vector, sample k values from the vector.
+  %   if an int, sample k values from 1:n.
+  % numItems : int
+  %   the number of items to sample.
+  % weights : vector
+  %   a weight vector, where the i'th index of ``weights`` is the 
+  %   weight for the i'th member of the population.
+
+  % if the user is trying to get multiple items from a single-integer array,
+  % assume that they want to sample between 1 and n
+  if isscalar(population) && numItems > 1
+    population = 1:population;
+  end
+
+  nargs = nargin;
+
+  if nargs > 3
+    error("Too many arguments.")
+  end
```

Comment:
Is this necessary given that you have specified the actual arguments in the function signature? I think an arguments block would work better here.